src/parser/ScriptParser.cpp

```diff
@@ -478,7 +475,7 @@ void ScriptParser::generateFunctionByteCode(ExecutionState& state, InterpretedCo
 
 #if defined(ENABLE_CODE_CACHE)
     CodeCache* codeCache = m_context->vmInstance()->codeCache();
-    bool cacheable = codeBlock->script()->isCacheable() && codeBlock->src().length() > CODE_CACHE_MIN_SOURCE_LENGTH;
+    bool cacheable = codeBlock->script()->isCacheable() && codeBlock->src().length() > codeCache->minSourceLength();
```

Comment:
IMO, for function caching, don't we have to check if this function is located in a eval code or module?
In the case of global code caching, we exclude any eval or module codes from caching.

Reply:
Good idea!

Reply:
I fixed it